addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/discovery/MaxDeviceDiscoveryService.java

```diff
@@ -91,7 +91,7 @@ public void onDeviceAdded(Bridge bridge, Device device) {
 
 	@Override
 	protected void startScan() {
-		// this can be ignored here as we discover via the bridge
+		if (maxCubeBridgeHandler != null) maxCubeBridgeHandler.clearDeviceList();

```

Comment:
Could you please add curly braces around the clearDeviceList() (event if it is just one call)? This makes debugging much easier.